contracts/gauges/LiquidityGaugeV2.vy

```diff
@@ -0,0 +1,720 @@
+
+    # Update user-specific integrals
+    _working_balance: uint256 = self.working_balances[addr]
+    self.integrate_fraction[addr] += _working_balance * (_integrate_inv_supply - self.integrate_inv_supply_of[addr]) / 10 ** 18
+    self.integrate_inv_supply_of[addr] = _integrate_inv_supply
+    self.integrate_checkpoint_of[addr] = block.timestamp
+
+
+@external
+def user_checkpoint(addr: address) -> bool:
+    """
+    @notice Record a checkpoint for `addr`
+    @param addr User address
+    @return bool success
+    """
+    assert (msg.sender == addr) or (msg.sender == self.minter)  # dev: unauthorized
+    self._checkpoint(addr)
+    self._update_liquidity_limit(addr, self.balanceOf[addr], self.totalSupply)
+    return True
+
+
+@external
+def claimable_tokens(addr: address) -> uint256:
+    """
+    @notice Get the number of claimable tokens per user
+    @dev This function should be manually changed to "view" in the ABI
+    @return uint256 number of claimable tokens per user
+    """
+    self._checkpoint(addr)
+    return self.integrate_fraction[addr] - Minter(self.minter).minted(addr, self)
+
+
+@external
+@nonreentrant('lock')
+def claimable_reward(_addr: address, _token: address) -> uint256:
+    """
+    @notice Get the number of claimable reward tokens for a user
+    @dev This function should be manually changed to "view" in the ABI
+         Calling it via a transaction will claim available reward tokens
+    @param _addr Account to get reward amount for
+    @param _token Token to get reward amount for
+    @return uint256 Claimable reward token amount
+    """
+    claimable: uint256 = ERC20(_token).balanceOf(_addr)
+    if self.reward_contract != ZERO_ADDRESS:
+        self._checkpoint_rewards(_addr, self.totalSupply)
+    claimable = ERC20(_token).balanceOf(_addr) - claimable
+
+    integral: uint256 = self.reward_integral[_token]
+    integral_for: uint256 = self.reward_integral_for[_token][_addr]
+
+    if integral_for < integral:
+        claimable += self.balanceOf[_addr] * (integral - integral_for) / 10**18
+
+    return claimable
+
+
+@external
+@nonreentrant('lock')
+def claim_rewards(_addr: address = msg.sender):
+    """
+    @notice Claim available reward tokens for `_addr`
+    @param _addr Address to claim for
+    """
+    self._checkpoint_rewards(_addr, self.totalSupply)
+
+
+@external
+@nonreentrant('lock')
+def claim_historic_rewards(_reward_tokens: address[MAX_REWARDS], _addr: address = msg.sender):
+    """
+    @notice Claim reward tokens available from a previously-set staking contract
+    @param _reward_tokens Array of reward token addresses to claim
+    @param _addr Address to claim for
+    """
+    for token in _reward_tokens:
+        if token == ZERO_ADDRESS:
+            break
+        integral: uint256 = self.reward_integral[token]
+        integral_for: uint256 = self.reward_integral_for[token][_addr]
+
+        if integral_for < integral:
```

Comment:
Can it ever be >?
What if it's `==`? For just one user?

Reply:
`>` should not be possible. `==` would be if the user tries to claim twice in the same block, or claims multiple times on a token where rewards have finished.